proxy/api/src/identity.rs

```diff
@@ -69,6 +69,18 @@ pub struct Metadata {
     pub ethereum: Option<Ethereum>,
 }
 
+impl Metadata {
+    pub fn update_payload(self, mut payload: PersonPayload) -> Result<PersonPayload, ExtError> {
```

Comment:
Something I'm thinking about is whether a `Metadata` that has `Option` fields might be more interesting for updates, i.e.
```rust
pub struct MetadataUpdate {
  name: Option<String>,
  ethereum: Option<Eth>,
}
```

Unfortunately this duplicates the structure (higher-kinded data anyone?!) but I think it's nice for targeting fields that you want to update (a "sensible" default is `None` on all fields). This could also allow the endpoint to have all those fields as optional so if they're missing then it just takes them as none and can set query parameters it cares about updating.

Let me know what you think.

Reply:
I've designed the proposed API with these problems in mind:
1. We need to be able to **remove** the Ethereum claim. How to express that, on JSON side and on Rust DTO side? `Option<Option<Eth>>`? Special meaning of a missing vs `null` JSON value? A list of removed values? They all are much trickier, especially considering communication between different JSON interpreters and different languages.
2. This structure is handled and interpreted in an identical way as the one passed to create an identity. It's easier to understand an API like this and requires less code to support, both on the frontend an the backend. 

There may be a better design and I'm not going to fight very hard for this one, but to me it seems reasonable.

Reply:
Right, that's a good point regarding being able to _remove_ the claim :thinking:

Reply:
That's a great point, sorry I missed that :)

I guess the correct structure would be the same as `Option` except it would convert `Update::Replace(Option<Eth>)` to `eth=null` and back again and the omission of `eth=` would be `Update::Dont`. But anyway, you're right this straight-forward and makes sense :)

_MAYBE_ what I want is lenses :P

Reply:
Lenses :heart:

Reply:
Getting back to this discussion, more specifically the design goal to be able to **remove** the Ethereum claim, note that this PR does _not_ fulfill that design goal. 

In [these lines below][https://github.com/radicle-dev/radicle-upstream/pull/1712/files#diff-ab07879e13206d820c0c01e08b337b8378c01698c2af0565710bbef5a21d81abR78-R80], we only set the `Ethereum` claim if `Metadata.ethereum` is `Some`.

I have added this PR to `radicle-link` to make it possible for us to remove an extension: https://github.com/radicle-dev/radicle-link/pull/582

Once that's in, and the latest revision of link is integrated in the Upstream, lines 78-80 should become:

```
match self.ethereum {
    Some(ethereum) => payload.set_ext(EthereumClaimExtV1::from(ethereum))?,
    None => payload.remove_ext::<EthereumClaimExtV1>()?,
}
```

Reply:
I think that you've already fixed that issue in `b3423759b2c298a7249811cb0aeebaf3e7da81e4`: `Support removing Ethereum claim ext` :thinking: I've added a test verifying that behavior.